### ISSUE TITLE
Bridgeless mode: Reflect, Proxy doesn't exist

### ISSUE THREAD

**Issue by farwayer** [facebook/react-native#43523]
### Description

After enabling bridgeless mode Reflect, Proxy became unavailable for some season

### Steps to reproduce

1. default project init
2. enabled new arch
3. enable bridgeless mode
4. add Reflect/Proxy reference: `new Proxy({}, {})` for example 

### React Native Version

0.73.6

### Affected Platforms

Runtime - Android, Runtime - iOS

### Areas

Bridgeless - The New Initialization Flow

### Output of `npx react-native info`

```text
System:
  OS: Linux
  CPU: x64 Intel
  Memory: 3.89 GB / 15.31 GB
Binaries:
  Node:
    version: 21.7.0
  Yarn:
    version: 1.22.21
  npm:
    version: 10.5.0
  Watchman: Not Found
SDKs:
  Android SDK:
    Android NDK: 25.2.9519653
IDEs:
  Android Studio: Not Found
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 3.0.6
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.6
    wanted: 0.73.6
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Stacktrace or Logs

```text
(NOBRIDGE) ERROR  ReferenceError: Property 'Proxy' doesn't exist

This error is located at:
    in Section (created by App)
    in RCTView (created by View)
    in View (created by App)
    in RCTView (created by View)
    in View (created by ScrollView)
    in RCTScrollView (created by ScrollView)
    in ScrollView (created by ScrollView)
    in ScrollView (created by App)
    in RCTView (created by View)
    in View (created by App)
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in AwesomeProject(RootComponent)
```

### Reproducer

https://github.com/farwayer/rn-bridgeless-no-proxy

**Comment by cortinico** [comment]
Thanks for reporting this @farwayer, we'll address this in 0.74

**Comment by mrousavy** [comment]
Not sure if this is supported by Hermes, but WeakRef also doesn't work.
I'm guessing this is either related to this issue, or just not yet implemented :)

(rn 0.74 rc)

**Comment by cortinico** [comment]
Closing as this has been fixed in 0.74. It's currently available in RC6
